apps/sms/js/thread_list_ui.js

```diff
@@ -339,27 +381,18 @@ var ThreadListUI = {
     }
   },
 
-  markReadUnread: function thlui_markReadUnread() {
-    var selected = this.selectionHandler;
-    var isRead = (this.readUnreadButton.dataset.action === 'mark-as-read');
+  markReadUnread: function thlui_markReadUnread(selected, isRead) {
+    if (typeof isRead === 'undefined') {
+      isRead = (this.readUnreadButton.dataset.action === 'mark-as-read');
+    }
```

Comment:
how about moving this to https://github.com/mozilla-b2g/gaia/pull/27615/files#diff-e2424773c46c9d5f4ff27c9c20b1bb0dR64 ?